tools/wasm-sourcemap.py

```diff
@@ -267,13 +296,7 @@ def main():
 
   code_section_offset = get_code_section_offset(wasm)
 
-  prefixes = []
-  for p in options.prefix:
-    if '=' in p:
-      prefix, replacement = p.split('=')
-      prefixes.append({'prefix': prefix, 'replacement': replacement})
-    else:
-      prefixes.append({'prefix': p, 'replacement': None})
+  prefixes = (Prefixes(options.prefix), Prefixes(options.load_prefix))
```

Comment:
So IIUC, previously we supported multiple prefixes which would all be substituted in the output source map, correct? And now we support just 2 prefixes: one that will be applied to the names in output source map, and another which will be applied to the names when we read them for embedding? And they are mutually exclusive?
I think that behavior should be made more clear in the help text, or docstring/comments or something. Also now rather than just having `prefixes[0]` and `prefixes[1]` it would be more clear if we just had 2 separate variables with different names.

Reply:
> ... [prefix] that will be applied to the names in output source map, and another which will be applied to the names when we read them for embedding? And they are mutually exclusive?

Correct. Not really, clang or rustc can generate debug file names that start with e.g. "/checkout/compiler/blah", which is not accessible from debugger point of view. The former ('sources') prefixes intended to re-point file name to some URL on the web, the latter ('load') allows to embed source into source map. Both prefixes can be applied if specified.


> having prefixes[0] and prefixes[1]

Replaced tuple with namedtuple

> should be made more clear in the help text

Help text changed